### PR TITLE
Tweak RichText

### PR DESCRIPTION
**Pull request by Ertanic**
Small tweak to allow tags to get object state from the game.

### CODE REVIEW

File: Robust.Client/UserInterface/RichText/MarkupTagManager.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ public sealed class MarkupTagManager
         new CommandLinkTag(),
         new FontTag(),
         new HeadingTag(),
-        new ItalicTag()
+        new ItalicTag(),
```
Comment (juliangiebel) [inline Robust.Client/UserInterface/RichText/MarkupTagManager.cs:27]: why add a comma here?

File: Robust.Client/UserInterface/RichText/MarkupTagManager.cs [hunk 2]
```diff
@@ -39,7 +39,7 @@ public sealed class MarkupTagManager
         typeof(CommandLinkTag),
         typeof(FontTag),
         typeof(HeadingTag),
-        typeof(ItalicTag)
+        typeof(ItalicTag),
```
Comment (juliangiebel) [inline Robust.Client/UserInterface/RichText/MarkupTagManager.cs:42]: same as above

File: Robust.Client/UserInterface/RichTextEntry.cs [hunk 3]
```diff
@@ -37,18 +37,37 @@ internal struct RichTextEntry
...
+        public RichTextEntry(FormattedMessage message, object tagsState, Control parent, MarkupTagManager tagManager, Type[]? tagsAllowed = null, Color? defaultColor = null)
+        {
+            _tagsState = tagsState;
+            Message = message;
+            _defaultColor = defaultColor ?? new(200, 200, 200);
+            _tagManager = tagManager;
+            _tagsAllowed = tagsAllowed;
+
+            Init(parent);
+        }
```
Comment (juliangiebel) [inline Robust.Client/UserInterface/RichTextEntry.cs:63]: Don't create a private init function and copy paste the constructor. Just add tagsState as another additional parameter

**Review by metalgearsloth — COMMENTED** [review]
I don't think storing any object on what is essentially text markup is a particularly good idea and see no good reason why paper isn't handling this.

File: Robust.Client/UserInterface/RichTextEntry.cs [hunk 4]
```diff
@@ -57,6 +76,10 @@ public RichTextEntry(FormattedMessage message, Control parent, MarkupTagManager
                 if (node.Name == null)
                     continue;
 
+                if (!node.Attributes.ContainsKey("state")
+                    && _tagsState != null)
+                    node.Attributes.Add("state", new(_tagsState));
+
```
Comment (juliangiebel) [inline Robust.Client/UserInterface/RichTextEntry.cs:82]: Don't save state like that inside the attributes. Pass it through using `MarkupDrawingContext`

File: Robust.Shared/Utility/MarkupNode.cs
Comment (juliangiebel) [inline Robust.Shared/Utility/MarkupNode.cs:1]: This is purely for the serialized data from the markup. I.e. `[testTag testattribute="example"]`

**Review by juliangiebel — CHANGES_REQUESTED** [review]
~Putting the state into the attributes like that isn't good.
Also `tagsState` shouldn't be a single object but rather something like a string object dictionary imo.~

### From the paper PR in content:
The way you're trying to use rich text here makes me think just using string replace <name> with the name, <job> with the job on finishing editing the paper would make much more sense.
RichText is at most supposed to show dynamic data that gets updated when "opening" the text (see the keybind tag)

**Pull request closed without merging**